### PR TITLE
Added documentation about recommended Kubernetes labels

### PR DESCRIPTION
**Pull request by ppatierno**
This PR adds some documentation around the recommended Kubernetes labels that we set on operands and cannot be overwritten by users when using templates.
This is related to https://github.com/strimzi/strimzi-kafka-operator/issues/9735

### CODE REVIEW

File: documentation/modules/configuring/ref-kubernetes-labels.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// assembly-config.adoc
+
+[id='ref-kubernetes-labels-{context}']
+= Kubernetes labels
+
+The Kubernetes documentation defines some https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[recommended labels] that could be set on the resources deployed within your application, in order to help describing objects in a way that all tools can understand.
+The Strimzi operator applies some of them to the operands it deploys:
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-kubernetes-labels.adoc:9]:
```suggestion
https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[Kubernetes labels] make it easier to organize, manage, and discover Kubernetes resources within your applications.
You can configure your own labels through the `template` configuration of Strimzi resources.
However, the Cluster Operator is responsible for applying the following Kubernetes labels to the operands it deploys. These labels cannot be overridden through configuration:
```

File: documentation/modules/configuring/ref-kubernetes-labels.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+* `app.kubernetes.io/name`: it is the component type within Strimzi which could be `kafka`, `zookeeper`, `cruise-control`, and so on.
+* `app.kubernetes.io/instance`: it is the name of the custom resource to which the operand belong to (that is, `my-cluster` as the name of the `Kafka` custom resource applied on the Kafka pods).
+* `app.kubernetes.io/part-of`: it is the same as the `app.kubernetes.io/instance` but with a `strimzi-` prefix.
+* `app.kubernetes.io/managed-by`: it defines which application is in charge of managing the operand like `strimzi-cluster-operator`, `strimzi-user-operator` and so on.
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-kubernetes-labels.adoc:14]:
```suggestion
* `app.kubernetes.io/name`: Identifies the component type within Strimzi, such as `kafka`, `zookeeper`,  and`cruise-control`.
* `app.kubernetes.io/instance`: Represents the name of the custom resource to which the operand belongs to. For instance, if a Kafka custom resource is named `my-cluster`, this label will bear that name on the associated pods.
* `app.kubernetes.io/part-of`: Similar to `app.kubernetes.io/instance`, but prefixed with `strimzi-`.
* `app.kubernetes.io/managed-by`: Defines the application responsible for managing the operand, such as `strimzi-cluster-operator` or `strimzi-user-operator`.
```

File: documentation/modules/configuring/ref-kubernetes-labels.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+.Example of Kubernetes labels on a Kafka Pod when deploying a `Kafka` custom resource named `my-cluster`
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-kubernetes-labels.adoc:16]:
```suggestion
.Example Kubernetes labels on a Kafka Pod when deploying a `Kafka` custom resource named `my-cluster`
```

File: documentation/modules/configuring/ref-kubernetes-labels.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  labels:
+    app.kubernetes.io/instance: my-cluster
+    app.kubernetes.io/managed-by: strimzi-cluster-operator
+    app.kubernetes.io/name: kafka
+    app.kubernetes.io/part-of: strimzi-my-cluster
+spec:
+  # ...
+----
+
+Those labels cannot be overridden by the user (that is, by using templates).
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-kubernetes-labels.adoc:32]:
```suggestion

```

File: documentation/modules/configuring/ref-kubernetes-labels.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// assembly-config.adoc
+
+[id='ref-kubernetes-labels-{context}']
+= Kubernetes labels
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/configuring/ref-kubernetes-labels.adoc:6]: Not sure about this title. I wanted something active and generally about labels, but also imply some restrictions. Might be okay

**Review by PaulRMellor — COMMENTED** [review]
Thanks Paolo. I've left some suggestions. I thought it might be better to describe the constraints earlier.

File: documentation/modules/configuring/ref-kubernetes-labels.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/configuring/ref-kubernetes-labels.adoc:6]: Isn't the title misleading by adding "Configuring"? It seems to suggest you can configure them while it's the opposite. Of course, I am not saying that "Kubernetes labels" is the best title, maybe there is something even better?

**Comment by PaulRMellor** [comment]
@ppatierno -- I wonder if we should move this section to sit as a sub-section under here instead? 
https://strimzi.io/docs/operators/in-development/deploying#assembly-customizing-kubernetes-resources-str

It's in the same chapter, but has content related to Kubernetes resources.

**Comment by ppatierno** [comment]
> It's in the same chapter, but has content related to Kubernetes resources.

@PaulRMellor I see your point but it's about "customizing" so still same fear like the title.

File: documentation/modules/configuring/ref-kubernetes-labels.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/configuring/ref-kubernetes-labels.adoc:6]: Yeah I like this title more. Thanks!

**Review by scholzj — APPROVED** [review]

**Comment by ppatierno** [comment]
@PaulRMellor I noticed your thumbs up on my latest comment about leaving the PR as it is. Are you going to approve as well? Or do you have any other comments?

**Review by im-konge — APPROVED** [review]

File: documentation/modules/configuring/ref-kubernetes-labels.adoc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[Kubernetes labels] make it easier to organize, manage, and discover Kubernetes resources within your applications.
+The Cluster Operator is responsible for applying the following Kubernetes labels to the operands it deploys.
+These labels cannot be overridden through `template` configuration of Strimzi resources:
+
+* `app.kubernetes.io/name`: Identifies the component type within Strimzi, such as `kafka`, `zookeeper`,  and`cruise-control`.
+* `app.kubernetes.io/instance`: Represents the name of the custom resource to which the operand belongs to. For instance, if a Kafka custom resource is named `my-cluster`, this label will bear that name on the associated pods.
+* `app.kubernetes.io/part-of`: Similar to `app.kubernetes.io/instance`, but prefixed with `strimzi-`.
+* `app.kubernetes.io/managed-by`: Defines the application responsible for managing the operand, such as `strimzi-cluster-operator` or `strimzi-user-operator`.
+
+.Example Kubernetes labels on a Kafka Pod when deploying a `Kafka` custom resource named `my-cluster`
```
Comment (PaulRMellor) [inline documentation/modules/configuring/ref-kubernetes-labels.adoc:17]:
```suggestion
.Example Kubernetes labels on a Kafka pod when deploying a `Kafka` custom resource named `my-cluster`
```

**Review by PaulRMellor — APPROVED** [review]
Thanks

**Comment by ppatierno** [comment]
The `TopicOperatorIT` test failure is not obviously related to the documentation changes. I am going to merge this.

**Pull request merged**